core/src/test/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/OpenSearchItemTest.kt

```diff
@@ -39,8 +40,26 @@ internal class OpenSearchItemTest {
     val intent = mockk<Intent>()
     every {
       anyConstructed<Intent>().putExtra(TAG_FILE_SEARCHED, searchListItem.value)
+        .putExtra(TAG_FILE_SEARCHED_NEW_TAB, false)
     } returns intent
-    OpenSearchItem(searchListItem).invokeWith(activity)
+    OpenSearchItem(searchListItem, false).invokeWith(activity)
+    verify {
+      activity.setResult(Activity.RESULT_OK, intent)
+      activity.finish()
+    }
+  }
+
+  @Test
+  fun `invoke with returns an Ok Result with list item value for new tab`() {
+    val searchListItem = RecentSearchListItem("")
+    val activity: AppCompatActivity = mockk()
+    mockkConstructor(Intent::class)
+    val intent = mockk<Intent>()
+    every {
+      anyConstructed<Intent>().putExtra(TAG_FILE_SEARCHED, searchListItem.value)
+        .putExtra(TAG_FILE_SEARCHED_NEW_TAB, true)
+    } returns intent
+    OpenSearchItem(searchListItem, true).invokeWith(activity)
     verify {
       activity.setResult(Activity.RESULT_OK, intent)
       activity.finish()
```

Comment:
I don't have much experience with unit testing, especially not with mocking and in Kotlin. Is this a sufficient test, or is it necessary to check whether a new tab was actually created?

Reply:
This is perfect, unit tests test `unit`s of our code, ie 1 logical branch, 1 method. This test class `OpenSearchItemTest` only cares about what `OpenSearchItem` does which is call `activity.setResult` with an intent and then call `activity.finish()`

The type of test you describe would be an Integration or (more likely) UI test, which have their place in the testing pyramid but I am not going to require here.

Are you working on the ViewModel test for the new action? It may look very complex but there should be examples to follow. If you have any questions I am here to help